Automattic/jetpack
# Untangle: show existing homepage when live-previewing another block theme

Related to:

- https://github.com/Automattic/dotcom-forge/issues/5309

This PR prevents live preview feature from always previewing the Index / Blog Home / Front Page template, when the site is part of the nav redesign.

Without this PR (in production), even if the current "Your homepage displays" setting is set to a static page, previewing a block theme will always preview the target theme's "home" template (Index / Blog Home / Front Page).

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






1. Enable this branch in Jetpack Beta Tester's WordPress.com Features. (Note that you must set define( 'JETPACK_MU_WPCOM_LOAD_VIA_BETA_PLUGIN', true ); to your wp-config.php)
2. Go to Settings -> Reading.
3. Set the "Your homepage displays" to a static page.
4. Go to Appearance -> Themes.
5. Install and Live Preview a block theme.
6. Verify that you are seeing the same static page (but in the target theme).
7. Turn off your A8C proxy (to test existing production behavior).
8. Make sure to not include the site in the early release list (`/wp-admin/options.php`, ensure `wpcom_classic_early_release` is unset)
9. Refresh the page
10. Verify that you are now seeing the theme's home template (Index / Blog Home / Front Page) (production behavior).

Appendix

"Live Preview" means this button (appears after you install the theme):

<img width="384" alt="image" src="https://github.com/Automattic/jetpack/assets/1525580/a6262323-b929-4321-afd8-2816814daf44">
